db/db_impl/db_impl_open.cc

```diff
@@ -1366,6 +1385,14 @@ Status DBImpl::Open(const DBOptions& db_options, const std::string& dbname,
     return s;
   }
 
+#ifdef OS_LINUX
+  auto db_paths = DbPaths(dbname, db_options, column_families);
+  s = FileSystem::Default()->HintDbPaths(db_paths);
```

Comment:
Can we always call env_->HintDbPaths(), and leave the implementation as empty?

Reply:
Added HintDbPaths to Env too